lib/improver/utilities/statistical_operations.py

```diff
@@ -0,0 +1,213 @@
+            ______/     \_________ Orogaphy
+
+        The orography heights are compared against the heights that
+        correspond with percentile values to find the band in which they
+        fall; this diagram hides the 2-dimensional variability of the snow
+        fall level. The percentile values are then interpolated to the
+        height of the point being considered. This constructs a
+        2-dimensional field of probabilities that snow will be falling at
+        each point in the orography field.
+    """
+
+    def __init__(self, reference_cube, output_name, inverse_ordering=False):
+        """
+        Initialise class.
+
+        Args:
+            reference_cube (iris.cube.Cube):
+                The percentiled field from which probabilities will be obtained
+                using the input cube.
+            output_name (str):
+                The name of the cube being created,
+                e.g.'probability_of_snowfall'.
+        Keyword Args:
+            inverse_ordering (bool):
+                Set True if the percentiled data increases in the opposite
+                sense to the percentile coordinate.
+                e.g.  0th Percentile - Value = 10
+                     10th Percentile - Value = 5
+                     20th Percenitle - Value = 0
+        """
+        self.reference_cube = reference_cube
+        self.output_name = output_name
+        self.inverse_ordering = inverse_ordering
+
+    def create_probability_cube(self, cube):
+        """
+        Create a probability cube in which to store the calculated
+        probabilities.
+
+        Args:
+            cube (iris.cube.Cube):
+                The input cube to be used as a template.
+        Returns:
+            probability_cube (iris.cube.Cube):
+                The new cube with suitable metadata.
+        """
+        cube_format = next(cube.slices([cube.coord(axis='y'),
+                                        cube.coord(axis='x')]))
+        probabilities = cube_format.copy(data=np.full(cube_format.shape, 1.,
+                                                      dtype=float))
+        percentile_coordinate, = [coord.name() for coord in
+                                  cube_format.coords()
+                                  if 'percentile' in coord.name()]
+        if percentile_coordinate:
+            probabilities.remove_coord(percentile_coordinate)
+        probabilities.units = 1
+        probabilities.rename(self.output_name)
+        return probabilities
+
+    def percentile_interpolation(self, cube, reference_cube):
+        """
+        Perform the percentile interpolation to construct the probability
+        field.
+
+        Args:
+            cube (iris.cube.Cube):
+                A cube of values, that effectively behave as thresholds, for
+                which it is desired to obtain probability values from a
+                percentiled reference cube.
```

Comment:
reference_cube needs to be added to Args although I think these would be better named cube -> threshold_cube, reference_cube -> percentile_cube